sparse_strips/vello_hybrid/src/image_cache.rs

```diff
@@ -0,0 +1,161 @@
+extern crate std;
+extern crate wgpu;
+
+use std::collections::HashMap;
+use wgpu::{BindGroup, BindGroupLayout, Device, Queue, Texture, TextureView};
+
+/// Represents an image resource for rendering
+#[derive(Debug)]
+pub struct ImageResource {
+    /// The texture containing the image data
+    pub texture: Texture,
+    /// The texture view for binding
+    pub view: TextureView,
+    /// The bind group for this image
+    pub bind_group: Option<BindGroup>,
+}
+
+/// Manages image resources for the renderer
+#[derive(Debug)]
+pub struct ImageCache {
+    /// Map of image IDs to resources
+    pub images: HashMap<u32, ImageResource>,
```

Comment:
We might be able to use a cheaper/faster collection type like using indices into a vector with free list or vector of `Occupied/Unoccupied` enums where `Unoccupied` slots point to the next free index.

Reply:
For the future, I have a work in progress crate where I was doing a resource pool using generational IDs and stuff based on how [Sokol](https://github.com/floooh/sokol) does them, but in Rust rather than C.

I plan to finish it up before RustWeek as I need it ... but happy to talk about this later.